samples/hello-cube.html

```diff
@@ -0,0 +1,313 @@
+<!DOCTYPE html>
+<html>
+<head>
+<meta name="viewport" content="width=600">
+<title>Web GPU Cube demo</title>
+<script src="gl-matrix-min.js"></script>
+</head>
+<body>
+<canvas height=1000 width=1000></canvas>
+<script>
+/*******
+* This preview of Web GPU uses WHLSL as its shading language. 
+* The choice of shading language(s) that will be ingested by the API is still under deliberation. 
+*******/
+const shader = `
+struct FragmentData {
+    float4 position : SV_Position;
+    float4 color : attribute(0);
+};
+
+vertex FragmentData vertex_main(float4 position : attribute(0),
+                                float4 color : attribute(1),
+                                constant float4x4* modelViewProjectionMatrix : register(b0)) {
```

Comment:
Why is the input a pointer to a float4x4?

Reply:
Do you mean as opposed to a reference or copy?

Reply:
As opposed to the more normal HLSL mechanism of defining a cbuffer/ConstantBuffer with your input variables.